pkg/stores/game/db.go

```diff
@@ -106,6 +109,60 @@ func (s *DBStore) Get(ctx context.Context, id string) (*entity.Game, error) {
 		g.WinnerIdx, g.LoserIdx, g.Request, g.History, sdata, s.gameEventChan, s.cfg)
 }
 
+// Similar to get but does not unmarshal the stats and timers and does
+// not play the game
+func (s *DBStore) GetQuickdata(ctx context.Context, id string) (*entity.Game, error) {
+	g := &game{}
+
+	if result := s.db.Where("uuid = ?", id).First(g); result.Error != nil {
+		return nil, result.Error
+	}
+
+	return FromStateQuickdata(g.Started, g.GameEndReason, g.Player0ID, g.Player1ID,
+		g.WinnerIdx, g.LoserIdx, g.Request, g.History, s.gameEventChan, s.cfg)
+}
+
+func (s *DBStore) GetRematchStreak(ctx context.Context, originalRequestId string) (*gs.GameInfoResponses, error) {
+	games := []*game{}
+	if results := s.db.Where("quickdata->>'s' = ?", originalRequestId).Order("updated_at desc").Find(games); results.Error != nil {
+		return nil, results.Error
+	}
+	return convertGamesToInfoResponses(games)
+}
+
+func (s *DBStore) GetRecentGames(ctx context.Context, playerId string, numGames int, offset int) (*gs.GameInfoResponses, error) {
+	games := []*game{}
+	if results := s.db.Limit(numGames).Offset(offset).Where("uuid = ?", playerId).Order("updated_at desc").Find(games); results.Error != nil {
+		return nil, results.Error
+	}
+	return convertGamesToInfoResponses(games)
+}
+
+func convertGamesToInfoResponses(games []*game) (*gs.GameInfoResponses, error) {
+	responses := []*gs.GameInfoResponse{}
+	for _, g := range games {
+		var mdata entity.Quickdata
+		err := json.Unmarshal(g.Quickdata.RawMessage, &mdata)
+		if err != nil {
+			return nil, err
+		}
+
+		playerInfo := []*gs.PlayerInfo{
+			&gs.PlayerInfo{UserId: g.Player0.UUID, Nickname: g.Player0.Username},
```

Comment:
I'm not so sure this will work. I think you need to specify that you're joining with the users table. Gorm is dumb. I'll test it though.